tests/densify/test_mvs_utils.py

```diff
@@ -124,6 +124,27 @@ def test_estimate_minimum_voxel_size(self) -> None:
         min_voxel_size = mvs_utils.estimate_minimum_voxel_size(points=points, scale=scale)
         self.assertAlmostEqual(min_voxel_size, 1 * scale, delta=0.1 * scale)
 
+    def test_compute_downsampling_psnr(self) -> None:
+        """Test the compute_downsampling_psnr function correctly produce the PSNR between two point clouds
+        We use the dummy original and downsampled point clouds as following:
+               o----o (1,1,1)
+             / |   /|
+            o--|--o |
+            | o --|-o                                 (1, 1/2, 1/2)
+            |/    |/                              o----o
+            o-- --o                      (0, 1/2, 1/2)
+        (0,0,0)
+        original point cloud              downsampled point cloud
+        """
+
+        original_point_cloud = np.array(
+            [[0, 0, 0], [0, 0, 1], [0, 1, 0], [1, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 0], [1, 1, 1]]
+        )
+        downsampled_point_cloud = np.array([[0, 0.5, 0.5], [1, 0.5, 0.5]])
+        psnr = mvs_utils.compute_downsampling_psnr(original_point_cloud, downsampled_point_cloud)
+
+        self.assertAlmostEqual(psnr, 8.36, 2)
```

Comment:
@codyly could you add an inline comment w/ the math behind this derivation? the values we plug into the equation etc

Reply:
I added some comments to explain the mathematics right below the description of the test case. @johnwlambert